nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -386,23 +383,34 @@ in
       '';
     };
 
-    systemd.services.slurmdbd = mkIf (cfg.dbdserver.enable) {
+    systemd.services.slurmdbd = let
+      # slurm strips the last component off the path
+      configPath = "$RUNTIME_DIRECTORY/slurmdbd.conf";
+    in mkIf (cfg.dbdserver.enable) {
       path = with pkgs; [ wrappedSlurm munge coreutils ];
 
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" "munged.service" "mysql.service" ];
       requires = [ "munged.service" "mysql.service" ];
 
-      # slurm strips the last component off the path
-      environment.SLURM_CONF =
-        if (cfg.dbdserver.configFile == null) then
-          "${slurmdbdConf}/slurm.conf"
-        else
-          cfg.dbdserver.configFile;
+      preStart = ''
+        cp ${slurmdbdConf} ${configPath}
+        chmod 600 ${configPath}
+        chown ${cfg.user} ${configPath}
+        ${optionalString (cfg.dbdserver.storagePassFile != null) ''
+          echo "StoragePass=$(cat ${cfg.dbdserver.storagePassFile})" \
+            >> ${configPath}
+        ''}
+      '';
+
+      script = ''
+        export SLURM_CONF=${configPath}
+        ${cfg.package}/bin/slurmdbd
+      '';
 
       serviceConfig = {
+        RuntimeDirectory = "slurmdbd";
         Type = "forking";
-        ExecStart = "${cfg.package}/bin/slurmdbd";
```

Comment:
you could keep this and add `environment.SLURM_CONF=configPath`

Reply:
The problem with `environment.SLURM_CONF is that `$RUNTIME_DIRECTORY` is not substituted but handed over verbatim.

Reply:
$RUNTIME_DIRECTORY is statically equal to /run/slurmdbd right ?

Reply:
Yes, it resolves to `/run/slurmdbd`.

Reply:
@symphorien Just to clarify: I initially tried with environment.SLURM_CONF but changed it to this solution so that i can use $RUNTIME_DIRECTORY in a consistent way. Directly writing `/run/slurmdbd` requires the assumption that systemd will put in under `/run`, which does not look like a clean solution to me.

Reply:
I was asking about removing the script and using bare execstart because with the script option the main process of the service is bash and this influences how systemd behaves (see the documentation of Type= in man systemd.service). Type=forking on with script is like Type=simple without because the shell forks to execute commands. But it's not that important either.

Reply:
ok, I see what you mean. I changed the type to simple and run the daemon in the foreground, which seems to be the preferred method upstream now.